src/overcooked_ai_py/mdp/layout_evaluator.py

```diff
@@ -30,14 +86,208 @@ def path_to_actions(path_0, path_1, terrain_mtx):
+    elif len(actions_1) == 0:
+        actions_1 = [(0, 0)] * len(actions_0)
+    # case where counter was used by both agents
+    else:
+        walk_graph = walk_graph_from_terrain(terrain_mtx)
+        # case where agent 0 is picking up from the counter
+        if actions_0[0] == 'interact':
+            temp_len = len(actions_0)
+
+            # calculate the action path to get the agent from where it ended previously to where it needs to pickup
+            action_pad = shortest_walk_path(walk_graph, prev_loc_0, counter_pickup_position_0, terrain_mtx)
+            pad_len = len(action_pad)
+
+            # remove last repetitive action if turning is not necessary
+            if pad_len > 1 and action_pad[pad_len-1] == action_pad[pad_len-2]:
+                action_pad = action_pad[:pad_len-1]
+
+            if len(action_pad) > len(actions_1):
+                actions_0 = action_pad + actions_0
+                actions_1 = actions_1 + [(0, 0)] * (temp_len + len(action_pad) - len(actions_1))
+            else:
+                actions_0 = [(0, 0)] * (len(actions_1) - len(action_pad)) + action_pad + actions_0
+                actions_1 = actions_1 + [(0, 0)] * temp_len
+        # case where agent 1 is picking up from the counter
+        elif actions_1[0] == 'interact':
+            temp_len = len(actions_1)
+
+            # calculate the action path to get the agent from where it ended previously to where it needs to pickup
+            action_pad = shortest_walk_path(walk_graph, prev_loc_1, counter_pickup_position_1, terrain_mtx)
+            pad_len = len(action_pad)
+
+            # remove last repetitive action if turning is not necessary
+            if pad_len > 1 and action_pad[pad_len - 1] == action_pad[pad_len - 2]:
+                action_pad = action_pad[:pad_len - 1]
+
+            if len(action_pad) > len(actions_0):
+                actions_1 = action_pad + actions_1
+                actions_0 = actions_0 + [(0, 0)] * (temp_len + len(action_pad) - len(actions_0))
+            else:
+                actions_1 = [(0, 0)] * (len(actions_0) - len(action_pad)) + action_pad + actions_1
+                actions_0 = actions_0 + [(0, 0)] * temp_len
+
+    assert len(actions_0) == len(actions_1), " resulting actions should have the same length. Please pad if otherwise"
+    return actions_0, actions_1
+
+def connect_action_path(action_paths):
+    """
+        Arguments:
+            action_paths (list of lists): a list of action paths to connect together
+        return:
+            connected action paths
+            currently just concatenates paths regardless of undefined actions
+        TODO: fill in undefined actions
+    """
+    total_path = []
+    for path in action_paths:
+        total_path += path
+    return total_path
+
+def calculate_entropy_of_path(path, ro):
+    """
+        Arguments:
+            path (list): a list of tuble for locations of agent 0
+            ro: parameter used in calculation of entropy
```

Comment:
I'm a bit confused as to why `ro` is being used here (usually spelt rho btw 😛). My understanding is that in this function, 𝜌 is only used to add a constant to the entropy calculation. Can't we just remove it?

Reply:
Thanks for letting me know its rho not ro :). Rho is a constant that is added to every sequence of consecutive identical actions. So for a path that consists purely of moving up, the rho entropy would only be added once. However for a path that consists of moving up and right with several turns, the rho entropy would be added several times. It is essentially a way to add in information about the number of turns in a given path.

Reply:
Do you know where you/Mesut got this formulation of path entropy from? I don't think I've seen it before

Reply:
I'm not sure where Mesut got it. I believe he may have formulated it himself.